.github/workflows/ci.yml

```diff
@@ -151,13 +151,20 @@ jobs:
         run: cd build && make flake8
       - id: files
         uses: Ana06/get-changed-files@v2.1.0
+        continue-on-error: true
       - name: clang-tidy
         run: |
+          if [[ -n "${{ steps.files.outputs.all }}" ]]; then
+            FILES="${{ steps.files.outputs.all }}"
+          else
+            git fetch origin main && git checkout -b main origin/main || true
```

Comment:
I think that checkout is unnecessary, we could run `git diff` right against `origin/main` after `git fetch` has run.

Reply:
'origin/main' may not exist if we push a new branch on the apache repository, but you are right that we should not checkout the main and run the clang tidy checks that afterwards. I fixed it by re-checking out the original commit in fd61850c10ec6a4597d9d7e56ded80e1bcb76942